ksqldb-engine/src/test/java/io/confluent/ksql/util/ExecutorUtilTest.java

```diff
@@ -102,6 +103,30 @@ public void shouldNotRetryOnNonRetriableException() throws Exception {
     assertThat(e.getMessage(), containsString("First non-retry exception"));
   }
 
+  @Test
+  public void shouldNotRetryOnCustomRetryableDenied() throws Exception {
+    // Expect
+    final AtomicBoolean firstCall = new AtomicBoolean(true);
+    final Callable<Object> throwsException = () -> {
+      if (firstCall.get()) {
+        firstCall.set(false);
+        // this is a retryable exception usually
+        throw new UnknownTopicOrPartitionException("First non-retry exception");
+      } else {
+        throw new RuntimeException("Test should not retry");
+      }
+    };
+
+    // When:
+    final RuntimeException e = assertThrows(
+        RuntimeException.class,
```

Comment:
The expectation is that we get the `UnknownTopicOrPartitionException` rather than `RuntimeException("Test should not retry")` right? Can we change this to `UnknownTopicOrPartitionException.class` to disambiguate?